### ISSUE TITLE
Code Highlighting Broken for Generic Functions with Default Values in .tsx Files

### ISSUE THREAD

**Issue by jinliming2** [microsoft/vscode#232221]
Does this issue occur when all extensions are disabled?: Yes

- Version: 1.95.0-insider
- Commit: 804f450ca900d24db25e7174e8b6dfb3fb2a318c
- Date: 2024-10-22T13:30:10.100Z
- Electron: 32.2.1
- ElectronBuildId: 10427718
- Chromium: 128.0.6613.186
- Node.js: 20.18.0
- V8: 12.8.374.38-electron.0
- OS: Darwin x64 24.0.0

**Describe the bug**
In VSCode, when writing a generic function with default values in a `.tsx` file, code highlighting is broken and colors are displayed incorrectly. The same code works fine in a `.ts` file.

**To Reproduce**
1. Open VSCode and create a new `.tsx` file.
2. Write the following code:
```tsx
const func = <T = string,>(args: T): string => {
  const t = typeof args;
  if (typeof t === 'string') {
    return t.split(',').join('_');
  }
  return '';
};
func('');

const add = (a: number, b: number) => {
  return a + b;
};
```
3. Observe the code highlighting.

**Expected behavior**
Code highlighting should work correctly and match the behavior in .ts files.

**Screenshots**
In `.tsx` file:
![Image](https://github.com/user-attachments/assets/9cceb590-fd21-467c-8ab7-2300095fee43)
Pay attention to the screenshot where the generic default parameter definitions, colons, and arrow function arrows are all highlighted in red, while the colors of keywords such as const, if, and return are incorrect.

In `.ts` file:
![Image](https://github.com/user-attachments/assets/381bc1ea-0829-41fb-8bc5-40846b928895)

**Comment by RedCMD** [comment]
workaround can be to remove the space before `=`
![Image](https://github.com/user-attachments/assets/a14556e1-205a-41e1-9c0c-299c032a0392)

**Comment by jinliming2** [comment]
> workaround can be to remove the space before `=` 

Yes, but not all scenarios allow removing =, for example:
```tsx
const func = <T = string,>(args?: T): T[] => {
  return args === undefined ? [] : [args];
};
const emptyStringArr = func();
const emptyNumberArr = func<number>();
```
![Image](https://github.com/user-attachments/assets/3dd23dd8-0042-4f92-a60d-d90b5a8335a2)

**Comment by RedCMD** [comment]
> but not all scenarios allow removing `=`

@jinliming2 removing the space ` `, not the equals
however TS will auto format the space back in

caused by `#jsx` being higher than `#arrow-function`
![Image](https://github.com/user-attachments/assets/34a11fb3-8084-4295-97f7-a9979dd39205)

**Comment by jinliming2** [comment]
@RedCMD Yes, it seems that removing the space after the generic parameter can eliminate the error, but the lint rule will automatically format it and add the space back. I have to disable the lint rule for this line.
